### PR TITLE
🌐 [i18n-KO] Translated `main_classes/data_collator.md` to Korean

### PR DESCRIPTION
**Pull request by fabxoe**
# What does this PR do?

Translated the `main_classes/data_collator.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [ ] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

@chhaewxn, @ahnjj, @jun048098, @fabxoe, @nuatmochoi, @heuristicwave

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/main_classes/data_collator.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 데이터 수집기(Data Collator)[[data-collator]]
```
Comment (nuatmochoi) [inline docs/source/ko/main_classes/data_collator.md:17]:
```suggestion
# 데이터 콜레이터(Data Collator)[[data-collator]]
```

File: docs/source/ko/main_classes/data_collator.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+데이터 콜레이터는 데이터셋 요소들의 리스트를 입력으로 사용하여 배치를 형성하는 객체입니다. 이러한 요소들은 `train_dataset` 또는 `eval_dataset의` 요소들과 동일한 타입 입니다. 배치를 구성하기 위해, 데이터 수집기는 (패딩과 같은) 일부 처리를 적용할 수 있습니다. [`DataCollatorForLanguageModeling`]과 같은 일부 수집기는 형성된 배치에 (무작위 마스킹과 같은) 일부 무작위 데이터 증강도 적용합니다. 사용 예시는 [예제 스크립트](../examples)나 [예제 노트북](../notebooks)에서 찾을 수 있습니다.
```
Comment (nuatmochoi) [inline docs/source/ko/main_classes/data_collator.md:19]:
```suggestion
데이터 콜레이터는 데이터셋 요소들의 리스트를 입력으로 사용하여 배치를 형성하는 객체입니다. 이러한 요소들은 `train_dataset` 또는 `eval_dataset의` 요소들과 동일한 타입 입니다. 배치를 구성하기 위해, 데이터 수집기는 (패딩과 같은) 일부 처리를 적용할 수 있습니다. [`DataCollatorForLanguageModeling`]과 같은 일부 콜레이터는 형성된 배치에 (무작위 마스킹과 같은) 일부 무작위 데이터 증강도 적용합니다. 사용 예시는 [예제 스크립트](../examples)나 [예제 노트북](../notebooks)에서 찾을 수 있습니다.
```

File: docs/source/ko/main_classes/data_collator.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## 기본 데이터 수집기[[transformers.default_data_collator]]
```
Comment (nuatmochoi) [inline docs/source/ko/main_classes/data_collator.md:22]:
```suggestion
## 기본 데이터 콜레이터[[transformers.default_data_collator]]
```

**Review by nuatmochoi — CHANGES_REQUESTED** [review]
glossary 참고하여 수정하였습니다:)

File: docs/source/ko/main_classes/data_collator.md (hunk 3 above)
Comment (fabxoe) [reply docs/source/ko/main_classes/data_collator.md:22]: 이번에 수집기라고 작성한 것은, 가짜연구소 아카데미 시기 때 철화님이 구글시트에 데이터 수집기라는 단어로 glossary에 남겨서 이렇게 해봤어요.
image

**Comment by ahnjj** [comment]
다른분들이 올려주신 리뷰외에 LGTM입니다!!!

File: docs/source/ko/_toctree.yml [hunk 4]
```diff
@@ -278,8 +278,8 @@
       title: (번역중) Callbacks
     - local: in_translation
       title: (번역중) Configuration
-    - local: in_translation
-      title: (번역중) Data Collator
+    - local: main_classes/data_collator
+      title: 데이터 수집기
```
Comment (fabxoe) [inline docs/source/ko/_toctree.yml:282]:
```suggestion
      title: 데이터 콜레이터
```

File: docs/source/ko/main_classes/data_collator.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 데이터 콜레이터(Data Collator)[[data-collator]]
+
+데이터 콜레이터는 데이터셋 요소들의 리스트를 입력으로 사용하여 배치를 형성하는 객체입니다. 이러한 요소들은 `train_dataset` 또는 `eval_dataset의` 요소들과 동일한 타입 입니다. 배치를 구성하기 위해, 데이터 수집기는 (패딩과 같은) 일부 처리를 적용할 수 있습니다. [`DataCollatorForLanguageModeling`]과 같은 일부 콜레이터는 형성된 배치에 (무작위 마스킹과 같은) 일부 무작위 데이터 증강도 적용합니다. 사용 예시는 [예제 스크립트](../examples)나 [예제 노트북](../notebooks)에서 찾을 수 있습니다.
```
Comment (fabxoe) [inline docs/source/ko/main_classes/data_collator.md:19]:
```suggestion
데이터 콜레이터는 데이터셋 요소들의 리스트를 입력으로 사용하여 배치를 형성하는 객체입니다. 이러한 요소들은 `train_dataset` 또는 `eval_dataset의` 요소들과 동일한 타입 입니다. 배치를 구성하기 위해, 데이터 콜레이터는 (패딩과 같은) 일부 처리를 적용할 수 있습니다. [`DataCollatorForLanguageModeling`]과 같은 일부 콜레이터는 형성된 배치에 (무작위 마스킹과 같은) 일부 무작위 데이터 증강도 적용합니다. 사용 예시는 [예제 스크립트](../examples)나 [예제 노트북](../notebooks)에서 찾을 수 있습니다.
```

**Review by stevhliu — APPROVED** [review]
Thank you, just need to resolve the merge conflict!

**Pull request merged**